kansha/board/comp.py

```diff
@@ -533,6 +535,59 @@ def export(self):
         e.body = f.getvalue()
         raise e
 
+    def _handle_column_delete(self, comp, action, source, data):
+        return self.delete_column(source[0])
+
+    def _handle_card_archive(self, comp, action, source, data):
+        self.archive_card(*source)
+        card_bo = source[-2]
+        scard = fts_schema.Card.from_model(card_bo.data)
+        self.search_engine.update_document(scard)
+        self.search_engine.commit()
+        self.set_reload_search()
+        notifications.add_event_to_history(self, (action, source, data))
+
+    def _handle_card_create(self, comp, action, source, data):
+        card_bo = source[-2]
+        scard = fts_schema.Card.from_model(card_bo.data)
+        self.search_engine.add_document(scard)
+        self.search_engine.commit()
+        self.set_reload_search()
+        notifications.add_event_to_history(self, (action, source, data))
+
+    def _handle_card_delete(self, comp, action, source, data):
+        card_id = data[0]
+        self.search_engine.delete_document(fts_schema.Card, card_id)
+        self.search_engine.commit()
+        self.set_reload_search()
+        notifications.add_event_to_history(self, (action, source, data))
+
+    def _handle_card_close(self, comp, action, source, data):
+        card_bo = source[-2]
+        scard = fts_schema.Card.from_model(card_bo.data)
+        self.search_engine.update_document(scard)
+        self.search_engine.commit()
+        self.set_reload_search()
```

Comment:
We would really use the events mechanism to avoid such code duplication. For example, when a card is deleted, we could emit both a `card_deleted` event and a generic `deleted` one. The `deleted` handler use `source[0]` to fetch the fts schema and execute the `search_engine.delete_document` call.

Or change the events to be an object instead of a tuple. So that the `card_deleted` event inherites from the `deleted` event